daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Anf.scala

```diff
@@ -208,62 +167,45 @@ private[lf] object Anf {
     case Right(binding) => target.SELocS(makeRelativeB(depth, binding))
   }
 
-  private[this] def flattenExp(depth: DepthA, env: Env, exp: source.SExpr)(
-      k: K[target.SExpr]
-  ): Res = {
-
-    transformExp(depth, env, exp)(k) { (_, sexpr) => k =>
-      Bounce { () =>
-        k(sexpr)
-      }
-    }
+  private[this] def flattenExp(depth: DepthA, env: Env, exp: source.SExpr): Res = {
+    transformExp(depth, env, exp) { (_, sexpr) => done(sexpr) }
   }
 
   private[this] def transformLet1(
       depth: DepthA,
       env: Env,
       rhs: source.SExpr,
       body: source.SExpr,
-  )(k: K[target.SExpr])(transform: Tx[target.SExpr]): Res = {
-
-    transformExp(depth, env, rhs)(k) { (depth, rhs) => k =>
+  )(transform: Tx[target.SExpr]): Res = {
+    transformExp(depth, env, rhs) { (depth, rhs) =>
       val depth1 = depth.incr(1)
       val env1 = trackBindings(depth, env, 1)
-      transformExp(depth1, env1, body) { body =>
-        Bounce { () =>
-          k(target.SELet1(rhs, body))
-        }
-      }(transform)
+      transformExp(depth1, env1, body)(transform) map { body =>
+        target.SELet1(rhs, body)
+      }
     }
   }
 
-  private[this] def flattenAlts(depth: DepthA, env: Env, alts0: List[source.SCaseAlt])(
-      k: K[List[target.SCaseAlt]]
-  ): Res = {
-
-    def loop(acc: List[target.SCaseAlt], alts: List[source.SCaseAlt]): Res = {
-      alts match {
-        case alt :: alts =>
-          flattenAlt(depth, env, alt) { alt =>
-            loop(alt :: acc, alts)
-          }
-        case Nil =>
-          k(acc.reverse)
-      }
-    }
-    loop(Nil, alts0)
+  private[this] def flattenAlts(
+      depth: DepthA,
+      env: Env,
+      alts0: List[source.SCaseAlt],
+  ): TailRec[List[SExpr.SCaseAlt]] = {
+    traverse(alts0, (alt: source.SCaseAlt) => flattenAlt(depth, env, alt))
   }
 
-  private[this] def flattenAlt(depth: DepthA, env: Env, alt: source.SCaseAlt)(
-      k: K[target.SCaseAlt]
-  ): Res = {
+  private[this] def flattenAlt(
+      depth: DepthA,
+      env: Env,
+      alt: source.SCaseAlt,
+  ): TailRec[SExpr.SCaseAlt] = {
 
     alt match {
       case source.SCaseAlt(pat, body) =>
         val n = patternNArgs(pat)
         val env1 = trackBindings(depth, env, n)
-        flattenExp(depth.incr(n), env1, body) { body =>
-          k(target.SCaseAlt(pat, body))
+        flattenExp(depth.incr(n), env1, body) map { body =>
```

Comment:
I do not like infix notation expect for operator.

Reply:
+ no real reason to use `{ }`

Reply:
Infix notation : fair enough. My scala is a bit rusty and I didn't remember if you could use dot notation along with braces for the method's argument, which I kept for the reason below.

I kept the braces because I they are everywhere in that file for lambdas so I went with that style. I could to break consistency with the rest of the file, or I could preserve consistency with the rest of the file but then I'd be touching places not affected by my change. Should I go ahead and replace braces everywhere? 

Sorry if my questions seem like bikeshedding, I am trying to get a sense of what's ok and what's not in PRs according to the team's culture. (For instance: is it ok to mix style and semantic changes in one PR? Do we value consistency within a file over global rules? Etc.)

Reply:
Changed map and flatMap to dot notation, kept the braces for consistency and for the reasons Nick mentioned.